rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -1971,6 +1984,7 @@ void main_window::CreateDockWindows()
 	m_debugger_frame->setObjectName("debugger");
 	m_log_frame = new log_frame(m_gui_settings, m_mw);
 	m_log_frame->setObjectName("logger");
+	m_kernel_explorer = new kernel_explorer(this);
```

Comment:
this whole idea of having a persistent window is wrong.
A: there is no need for it to be created on boot
B: 99% of the users won't need it
C: it will always show up as a hidden window if you investigate your open windows

The correct way to do it is to make it non-modal and to use a singleton approach as seen in GalCiv's code (e.g. the skylanders dialog).

This also removes the need of a convoluted signal slot mechanic (which you overcomplicated by a lot, since there are already native signals for this)

Reply:
What about now?